### PR TITLE
Reference implementation: Adding with sharing to checkout integration classes.

### PR DESCRIPTION
**Pull request by KalyanBoya**
### What does this PR do?
Updating the reference implementation to have 'withsharing' keyword
### What issues does this PR fix or reference?

@W-12214313@
@https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001HP6zfYAD/view@
### Functionality Before
Checkout did not had strict record level access checks

### Functionality After
This would enforce record level access checks for checkout integrations.

### How to Test/Testing Effort 
run the checkout

### CODE REVIEW

**Comment by KalyanBoya** [comment]
@gurpreetsainisalesforce @unandyala please review

File: examples/b2c/checkout/integrations/classes/B2CDeliverySample.cls [hunk 1]
```diff
@@ -247,49 +247,12 @@ global class B2CDeliverySample implements sfdc_checkout.CartShippingCharges {
...
-                if (existingodm.ProductId == null || existingodm.Carrier == null || existingodm.ClassOfService == null) {
-                    existingodm.ProductId = product2IdForThisDeliveryMethod;
-                    existingodm.Carrier = shippingOption.serviceName;
-                    existingodm.ClassOfService = shippingOption.provider;
-                    update(existingodm);
-                }
-                orderDeliveryMethodIds.add(existingodm.Id);
-            }
-            i+=1;
+        List<OrderDeliveryMethod> odms = [SELECT Id, ProductId, Carrier, ClassOfService, IsActive FROM OrderDeliveryMethod Limit 2 ];
```
Comment (unandyala) [inline examples/b2c/checkout/integrations/classes/B2CDeliverySample.cls:251]: Could you explain what is the change in the functionality with these code changes?

Comment (KalyanBoya) [reply examples/b2c/checkout/integrations/classes/B2CDeliverySample.cls:251]: For the guest users we should not allow them to create the order delivery method. Now we removed the create operation and only reading it from DB.

**Review by gurpreetsainisalesforce — DISMISSED** [review]

**Review by unandyala — DISMISSED** [review]

**Review by tarcang — DISMISSED** [review]

File: examples/b2c/checkout/integrations/classes/B2CDeliverySample.cls [hunk 2]
```diff
@@ -247,49 +247,12 @@ global class B2CDeliverySample implements sfdc_checkout.CartShippingCharges {
     }
 
     private List<Id> getOrderDeliveryMethods(List<ShippingOptionsAndRatesFromExternalService> shippingOptions) {
-        String defaultDeliveryMethodName = 'Order Delivery Method';
```
Comment (szelenin-salesforce) [inline examples/b2c/checkout/integrations/classes/B2CDeliverySample.cls:250]: @KalyanBoya seems like we have same class in B2B: https://github.com/forcedotcom/commerce-on-lightning/blob/develop/examples/b2b/checkout/integrations/classes/B2BDeliverySample.cls

**Pull request closed without merging**